rpc/ws.go

```diff
@@ -62,8 +62,8 @@ func (s *wsServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			break
 		}
 
-		var req *rpctypes.Request
-		err = json.Unmarshal(message, &req)
+		req := new(rpctypes.Request)
+		err = vjson.UnmarshalStruct(message, req)
```

Comment:
The validation function expects a pointer to a struct as its top-level parameter, not a pointer to a pointer to a struct. We had discussed changing this previously, because most code examples you see don't push allocation into the Unmarshaller like this. They are all updated now.